File: src/common/py/data/entities/project/logbook/project_logbook_record.py
import abc
import time
from dataclasses import dataclass, field

from dataclasses_json import dataclass_json

RecordID = int


@dataclass_json
@dataclass
class ProjectLogbookRecord(abc.ABC):
    """
    Base class for all project logbook records.

    Attributes:
        record: The record entry ID.
        timestamp: The timestamp of the record.
        seen: Whether the record has been seen by the user.
    """

    record: RecordID = 0
    timestamp: float = field(default_factory=lambda: time.time())

    seen: bool = False

File: src/common/web/data/entities/project/logbook/ProjectLogbookRecord.ts
/**
 * A single record of a project's logbook.
 *
 * @param record - The record entry ID.
 * @param timestamp - The timestamp of the record.
 * @param seen - Whether the record has been seen by the user.
 */
export class ProjectLogbookRecord {
    public readonly record: number;
    public readonly timestamp: number;

    public readonly seen: boolean;

    public constructor(record: number, timestamp: number, seen: boolean = false) {
        this.record = record;
        this.timestamp = timestamp;

        this.seen = seen;
    }
}
